http-server/src/Main.gren

```diff
@@ -0,0 +1,75 @@
+module Main exposing (main)
+
+import Node
+import Bytes exposing (Bytes)
+import Bytes.Encode as BE
+import Node.Program as Program exposing (Program)
+import Node.Http as Http
+
+
+main : Program Model Msg
+main =
+    Program.define
+        { init = init
+        , update = update
+        , subscriptions = subscriptions
+        }
+
+
+type alias Model =
+    { serverKey : Http.ServerKey
+    }
+
+
+type Msg
+    = HandleRequest Http.Request
+    {-
+        type alias Request =
+            { headers : Dict String String
+            , body : Bytes
+            , url : Url
+            , method : Method
+            }
+        
+        type Method
+            = GET
+            | HEAD
+            | POST
+            | PUT
+            | DELETE
+            | CONNECT
+            | TRACE
+            | PATCH
+    -}
+
+
+init : Program.AppInitTask { model : Model, command : Cmd Msg }
+init =
+    Program.await Http.createServer <| \serverKey ->
```

Comment:
I just realized that starting servers can fail (common case being that the port is not available), and so `createServer` should probably be a `Task Http.CreateServerError Http.Server`.

Instead, perhaps there should be a `Http.initialize` function that returns a permission value that is required for creating servers?

Feels a bit boilerplatey, but it's possible to create a wrapper function around Program that sets up some simple error handling for when creating a server doesn't work.

Reply:
I agree. But it's not clear to me how a task that can fail can be handled in init. Should `createServer` return a `Cmd msg` that's handled in update?

Reply:
Sorry if that question doesn't make much sense. Still sometimes get a little confused about the relationship of commands and tasks...

Reply:
Ok I think I got it 😅 

`Http.createServer` returns a task, which you can pass to `Task.attempt` to get a command, which lets you handle the potential `Err` in `update`.